### PR TITLE
feat/tree: add tree chart type

### PR DESCRIPTION
**Pull request by AndreZiviani**
**Tree**

image

**Example**

```go
package main

import (
	"os"
        "io"

	"github.com/go-echarts/go-echarts/v2/charts"
	"github.com/go-echarts/go-echarts/v2/components"
	"github.com/go-echarts/go-echarts/v2/opts"
)

func main() {
    tree := charts.NewTree()
    tree.SetGlobalOptions(
        charts.WithTitleOpts(opts.Title{Title: "Tree Chart"}),
            charts.WithToolboxOpts(opts.Toolbox{
                Show: true,
                    Feature: &opts.ToolBoxFeature{
                        SaveAsImage: &opts.ToolBoxFeatureSaveAsImage{
                            Show:  true,
                            Type:  "png",
                            Title: "Download as PNG",
                        },
                    },
                }),
                charts.WithLegendOpts(opts.Legend{Show: true}),
                charts.WithTooltipOpts(opts.Tooltip{Show: true}),
	)

    chield1 := &opts.TreeData{
        Name: "test2",
        Value: 2,
    }
    chield2 := &opts.TreeData{
        Name: "test3",
        Value: 3,
    }

    children := make([]*opts.TreeData, 0)
    children = append(children, chield1)
    children = append(children, chield2)

    nodes := make([]*opts.TreeData, 0)
    nodes = append(nodes,
        opts.TreeData{
            Name: "test",
            Value: 1,
            Children: children,
        },
    )

    tree.AddSeries("tree", nodes).
        SetSeriesOptions(
            charts.WithTreeOpts(
                opts.TreeChart{
                    Layout: "orthogonal",
                    Orient: "LR",
                },
            ),
            charts.WithLabelOpts(opts.Label{Show: true, Position: "top", Color: "Black"}),
        )
    page := components.NewPage()
    page.AddCharts(tree)
    f, _ := os.Create("tree.html")
    page.Render(io.MultiWriter(f))
}
```

### CODE REVIEW

File: types/charts.go [hunk 1]
```diff
@@ -26,4 +26,5 @@ const (
 	ChartSurface3D     = "surface"
 	ChartThemeRiver    = "themeRiver"
 	ChartWordCloud     = "wordCloud"
+	Tree               = "tree"
```
Comment (chenjiandongx) [inline types/charts.go:29]: Tree -> ChartTree will be a better choice

**Review by Koooooo-7 — DISMISSED** [review]

File: charts/tree.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+import (
+	"github.com/go-echarts/go-echarts/v2/opts"
+	"github.com/go-echarts/go-echarts/v2/render"
+	"github.com/go-echarts/go-echarts/v2/types"
+)
+
+// Tree represents a Tree chart.
+type Tree struct {
+	RectChart
```
Comment (chenjiandongx) [inline charts/tree.go:11]: Should be `BaseConfiguration` here

**Comment by Koooooo-7** [comment]
Could u plz add an example at [here](https://github.com/go-echarts/examples) also.

**Review by chenjiandongx — APPROVED** [review]
LGTM

**Pull request merged**